### PR TITLE
[doc] Add Zero Grad `set_to_none=True` trick

### LINKED ISSUE #6534: Option of doing optimizer.zero_grad(set_to_none=True).
## 🚀 Feature

An option to set gradients to None instead of zero. This was introduced in PyTorch 1.7. [Documentation](https://pytorch.org/docs/master/optim.html#torch.optim.Optimizer.zero_grad).
### Motivation
Training Speed improvements, as discussed by PyTorch [here](https://pytorch.org/tutorials/recipes/recipes/tuning_guide.html#use-parameter-grad-none-instead-of-model-zero-grad-or-optimizer-zero-grad). 
### Pitch

I changed Line 1400 of https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/core/lightning.py. 
From 
```    
def optimizer_zero_grad(self, epoch: int, batch_idx: int, optimizer: Optimizer, optimizer_idx: int):  
    optimizer.zero_grad()
```
To 
```    
def optimizer_zero_grad(self, epoch: int, batch_idx: int, optimizer: Optimizer, optimizer_idx: int):
     optimizer.zero_grad(set_to_none=True)
```
I achieved a modest speedup in my training. In my case of 4096 image epoch, batchsize 32.   
My times before the change were **41.42s**, and after were **41.05s**. Measured as the min over three runs.

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

This section add a description about set_to_none training trick.

Fixes  #6534

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by kaushikb11 — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

File: docs/source/benchmarking/performance.rst [hunk 1]
```diff
@@ -181,3 +181,21 @@ Most UNIX-based operating systems provide direct access to tmpfs through a mount
...
+In order to modestly improve performance, once can override :func:`~pytorch_lightning.core.lightning.LightningModule.optimizer_zero_grad`
+
+For a more detailed explanation of pro / cons of this technique,
+read `this <https://pytorch.org/docs/master/optim.html#torch.optim.Optimizer.zero_grad>`__ documentation by the PyTorch team.
+
+.. testcode::
+
+    import pytorch_lightning as pl
+
+    class Model(pl.LightningModule)
```
Comment (rohitgr7) [inline docs/source/benchmarking/performance.rst:198]:
```suggestion
In order to modestly improve performance, once can override :meth:`~pytorch_lightning.core.lightning.LightningModule.optimizer_zero_grad`.

For a more detailed explanation of pros / cons of this technique,
read `this <https://pytorch.org/docs/master/optim.html#torch.optim.Optimizer.zero_grad>`_ documentation by the PyTorch team.

.. testcode::

    class Model(LightningModule)
```

**Review by rohitgr7 — APPROVED** [review]
does it resolve https://github.com/PyTorchLightning/pytorch-lightning/issues/6534?

**Comment by tchaton** [comment]
> does it resolve #6534?

Yes, it does. Good catch !

Best,
T.C

**Review by ananthsub — APPROVED** [review]

File: docs/source/benchmarking/performance.rst [hunk 2]
```diff
@@ -181,3 +181,19 @@ Most UNIX-based operating systems provide direct access to tmpfs through a mount
...
+In order to modestly improve performance, once can override :meth:`~pytorch_lightning.core.lightning.LightningModule.optimizer_zero_grad`.
+
+For a more detailed explanation of pros / cons of this technique,
+read `this <https://pytorch.org/docs/master/optim.html#torch.optim.Optimizer.zero_grad>`_ documentation by the PyTorch team.
+
+.. testcode::
+
+    class Model(LightningModule):
+
+        def optimizer_zero_grad(self, epoch: int, batch_idx: int, optimizer: Optimizer, optimizer_idx: int):
```
Comment (rohitgr7) [inline docs/source/benchmarking/performance.rst:198]:
```suggestion
        def optimizer_zero_grad(self, epoch, batch_idx, optimizer, optimizer_idx):
```
